modules/civicrm_member_roles/civicrm_member_roles.module

```diff
@@ -593,46 +593,44 @@ function _civicrm_member_roles_sync($ext_uid = NULL, $cid = NULL, $sync_type = N
     $contactMemberships = CRM_Utils_Array::value('values', $memberships);
     $addRoles           = array();
     $expRoles           = array();
-
-    $roleCondition = is_array($rid) ? 'IN' : '=';
-
-    if (empty($contactMemberships) && !empty($rid)) {
-      db_delete('users_roles')->condition('uid', $uid)->condition('rid', $rid, $roleCondition)->execute();
-    }
```

Comment:
is this no longer happening now? I just replicated the following problem which wasn't happening with the previous code.

- Create 2 member sync rule. Eg role1 => memtype1 & role2 => memtype2.
- Create a contact (with drupal user) and add a membership of type = memtype1.
- Sync the rule. role1 is added to the user (Correct).
- Delete the membership `memtype1` from the contact and add the membership of type = `memtype2`.
- Sync the rule again. role2 is added (Correct). But role1 is not removed from the contact (Incorrect).

As `memtype1` is not associated with the civi contact, role1 should have been removed after the sync. It was working as per the above condition.

@sadashivdalvi